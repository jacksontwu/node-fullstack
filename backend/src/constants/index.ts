import { join } from 'path';

const logDir = join(process.cwd(), 'logs');

export { logDir };
