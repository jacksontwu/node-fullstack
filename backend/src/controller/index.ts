import healthController from './health-controller';

export { healthController };
