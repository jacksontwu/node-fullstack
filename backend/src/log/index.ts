import config from './log4js';
import { errLogger, resLogger, reqLogger } from './logger';

export { config, errLogger, reqLogger, resLogger };
