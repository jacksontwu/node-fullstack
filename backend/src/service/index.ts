import HealthService from './health-service';

export { HealthService };
