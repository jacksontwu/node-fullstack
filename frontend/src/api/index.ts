import request from '../utils/request';

export const fetchData = (query: any) => {
    return request({
        url: './table.json',
        method: 'get',
        params: query,
    });
};
