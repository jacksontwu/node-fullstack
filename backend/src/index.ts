import * as Koa from 'koa';
import * as Router from 'koa-router';
import * as  cors from '@koa/cors';
import * as bodyParser from 'koa-bodyparser';
import { PORT } from './config';
import AppRoutes from './routes';
import { configure } from 'log4js';
import { config, reqLogger, resLogger } from 'src/log';

const app = new Koa();
const router = new Router();

AppRoutes.forEach((route) => router[route.method](route.path, route.action));

configure(config);
app.use(async (ctx, next) => {
    const start = new Date();
    reqLogger(ctx);
    await next();
    const end = new Date();
    resLogger(ctx, end.getTime() - start.getTime());
});

app.use(cors());
app.use(bodyParser());
app.use(router.routes());
app.use(router.allowedMethods());

app.listen(PORT);

console.log(`Backend start successfully, port:${PORT}`);
