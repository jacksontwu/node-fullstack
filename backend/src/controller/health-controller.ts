import { HealthService } from 'src/service';

class HealthController {
    private service: HealthService = new HealthService();

    health = async (ctx) => {
        const health = this.service.health();
        ctx.body = { health };
    };
}

export default new HealthController();
