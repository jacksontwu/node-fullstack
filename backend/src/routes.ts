import { healthController } from './controller';

export default [
    {
        path: '/health',
        method: 'get',
        action: healthController.health,
    },
];
