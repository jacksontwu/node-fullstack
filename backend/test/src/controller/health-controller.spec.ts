import { healthController } from '../../../src/controller';

describe('healthController', () => {
    it('health', async () => {
        const ctx = {
            body: null,
        };
        await healthController.health(ctx);
        expect(ctx.body).toMatchSnapshot();
    });
});
