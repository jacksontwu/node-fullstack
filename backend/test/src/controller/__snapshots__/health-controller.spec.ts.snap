// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`healthController health 1`] = `
Object {
  "health": "success",
}
`;
